cg/meta/upload/scout/uploadscoutapi.py

```diff
@@ -155,64 +138,41 @@ def get_splice_junctions_bed(self, case_id: str, sample_id: str) -> Optional[hk_
         return splice_junctions_bed
 
     def get_rna_coverage_bigwig(self, case_id: str, sample_id: str) -> Optional[hk_models.File]:
-        """Get a rna coverage bigwig file for case in housekeeper
+        """Get a RNA coverage bigwig file for case in housekeeper."""
 
-        Args:
-            case_id     (string):       Case identifier
-            sample_id   (string):       Sample identifier
-        Returns:
-            File in housekeeper (Optional[hk_models.File])
-        """
+        tags: Set[str] = {"coverage", "bigwig", sample_id}
 
-        # This command can be executed as:
-        # ´housekeeper get file -V --tag coverage --tag bigwig <sample_id>´
-        tags: {str} = {"coverage", "bigwig", sample_id}
+        return self.housekeeper.find_file_in_latest_version(case_id=case_id, tags=tags)
 
-        rna_coverage_bigwig: Optional[
-            hk_models.File
-        ] = self.housekeeper.find_file_in_latest_version(case_id=case_id, tags=tags)
+    def get_unique_dna_cases_related_to_rna_case(self, case_id: str) -> Set[str]:
+        """Return a set of unique dna cases related to a RNA case"""
+        case_obj: models.Family = self.status_db.family(case_id)
 
-        return rna_coverage_bigwig
+        return {
+            dna_case
+            for dna_case_dict in self.create_rna_dna_sample_case_map(rna_case=case_obj).values()
+            for case_list in dna_case_dict.values()
+            for dna_case in case_list
```

Comment:
```suggestion
            for dna_case_dict in self.create_rna_dna_sample_case_map(rna_case=case_obj).values()
            for case_list in dna_case_dict.values()
            for dna_case in case_list
```
This triple for loop with a dict comprehension looks rather daunting and fragile. Would be great to split it up to make each part more atomic and easier to follow

Reply:
Well, it is a nested dictionary that is returned. The functions inside that function are all tested and I could make another test here as well - in all honesty, I don't know if we can with our current database logic make this any better, and the issue really lies in the fact how downstream functions were defined due to that database logic.